arctic/store/_ndarray_store.py

```diff
@@ -163,6 +158,41 @@ def get_info(self, version):
         ret['rows'] = int(version['up_to'])
         return ret
 
+    def _chunked_read(self, arctic_lib, version, symbol, date_range, read_preference=None):
+        collection = arctic_lib.get_top_level_collection()
+        if read_preference:
+            collection = collection.with_options(read_preference=read_preference)
+        return self._do_chunked_read(collection, version, symbol, date_range)
+
+    def _do_chunked_read(self, collection, version, symbol, date_range):
+        segment_count = version.get('segment_count', None)
+
+        spec = {'symbol': symbol,
+                'parent': version.get('base_version_id', version['_id']),
+                }
+
+        if date_range is not None:
+            spec['segment'] = date_range.mongo_query()
+
+        segments = []
+        i = -1
+        for i, x in enumerate(collection.find(spec, sort=[('segment', pymongo.ASCENDING)],)):
+            segments.append(decompress(x['data']))
+
+        data = b''.join(segments)
+
+        # we can retrieve any number of segments due to how we chunk and query the chunks
+        # so anything from 0 to segment_chunks (inclusive) is ok
+        if segment_count is not None and i + 1 > segment_count:
```

Comment:
The size chunked code was trying to assert, in the presence of `append` that the returned number of documents matches the `segment_count` expected.  

This condition looks sufficiently weaker that perhaps we delete - I can't think of a mechanism by which more documents would come back than we expected.